wrapanapi/systems/redfish.py

```diff
@@ -4,27 +4,124 @@
 """
 from __future__ import absolute_import
 
+import redfish_client
+
+from wrapanapi.entities.server import Server, ServerState
 from wrapanapi.systems.base import System
 
 
+class RedfishServer(Server):
+    state_map = {
+        'On': ServerState.ON,
+        'Off': ServerState.OFF,
+        'PoweringOn': ServerState.POWERING_ON,
+        'PoweringOff': ServerState.POWERING_OFF,
+    }
+
+    def __init__(self, system, raw=None, **kwargs):
+        """
+        Constructor for RedfishServer.
+
+        Args:
+            system: RedfishSystem instance
+            raw: the root resource in the Redfish API
+            odata_id: (optional) the @odata.id reference of this instance
+        """
+        self._odata_id = raw['@odata.id'] if raw else kwargs.get('odata_id')
+        if not self._odata_id:
+            raise ValueError("missing required kwargs: 'odata_id'")
+
+        super(RedfishServer, self).__init__(system, raw, **kwargs)
+
+    @property
+    def server_cores(self):
+        """Return the number of cores on this server."""
+        return sum([int(p.TotalCores) for p in self.raw.Processors.Members])
+
+    @property
+    def server_memory(self):
+        """Return the amount of memory on the server, in MiB."""
+        return self.raw.MemorySummary.TotalSystemMemoryGiB * 1024
+
+    @property
+    def state(self):
+        """Retrieve the current power status of the physical server."""
+        return self.raw.PowerState
+
+    def _get_state(self):
```

Comment:
We don't need this function in subclasses, please remove (see my suggestion above).